src/main/scala/com/rbmhtechnology/calliope/serializer/kafka/PayloadFormatSerializer.scala

```diff
@@ -48,22 +51,46 @@ class PayloadFormatSerializer[A <: AnyRef] private(serializer: PayloadSerializer
 }
 
 object PayloadFormatDeserializer {
-  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
-    new PayloadFormatDeserializer[A](DelegatingStringManifestPayloadSerializer(system), payloadMapper)
 
   def apply()(implicit system: ActorSystem): PayloadFormatDeserializer[AnyRef] =
     apply[AnyRef](identity)
 
-  def create[A](payloadMapper: JFunction[AnyRef, A], system: ActorSystem): PayloadFormatDeserializer[A] =
-    apply(payloadMapper.apply)(system)
+  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
+    instance(payloadMapper, err => throw err)
+
+  def option[A](payloadMapper: PartialFunction[AnyRef, A])(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
+    apply(payloadMapper.lift)
+
+  def filterError[A](payloadMapper: PartialFunction[AnyRef, A], onError: Throwable => Unit)(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
```

Comment:
Ok, I agree on using vavr then, as it will make the Java API more closely resemble the scala API. Will go for the Java function:

`createOptionFilterError[A](payloadMapper: JFunction[AnyRef, JOption[A]], system: ActorSystem): PayloadFormatDeserializer[vavr.Try[JOption[A]]]`

As for the proposed two versions of `attemptFilterError` (will rename it accordingly to `optionFilterError`) what do you think?